packages/ember/src/components/form/select.gts

```diff
@@ -0,0 +1,433 @@
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { get, action } from '@ember/object';
+import Component from '@glimmer/component';
+// @ts-expect-error Glimmer doesn't currently ship a type for the `cached` decorator
+// https://github.com/glimmerjs/glimmer.js/issues/408
+import { tracked, cached } from '@glimmer/tracking';
+// @ts-expect-error Ember keyboard doesn't currently ship a type for the `on-key` modifier
+import onKey from 'ember-keyboard/modifiers/on-key';
+import { runTask } from 'ember-lifeline';
+
+import BoundValue from './bound-value.ts';
+import onInsert from '../../modifiers/did-insert.ts';
+
+import type { Optional } from '../../types.d.ts';
+
+declare type SelectOption<T> = {
+  label: string;
+  value: string | T;
+  raw: T;
+};
+
+interface SelectItemSignature<T> {
+  Args: {
+    optionIndex: number;
+    activeIndex: number;
+    currentValue: string | T;
+    option: SelectOption<T>;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLLIElement;
+}
+
+class SelectItem<T> extends Component<SelectItemSignature<T>> {
+  get classList() {
+    let classes = ['dropdown-item'];
+
+    const useIndexActive = this.args.activeIndex != -1;
+    const isCurrentIndex = this.args.optionIndex === this.args.activeIndex;
+    const isCurrentValue = this.args.option.value === this.args.currentValue;
+
+    if (useIndexActive && isCurrentIndex) {
+      classes.push('active');
+    }
+
+    if (!useIndexActive && isCurrentValue) {
+      classes.push('active');
+    }
+
+    return classes.join(' ');
+  }
+
+  get isActive() {
+    return this.args.option.value === this.args.currentValue;
+  }
+
+  <template>
+    <li
+      class={{this.classList}}
+      role="option"
+      aria-selected={{this.isActive}}
+      ...attributes
+    >
+      {{yield}}
+    </li>
+  </template>
+}
+
+export interface SelectSignature<T> {
+  Args: {
+    disabled?: boolean;
+    loading?: boolean;
+    scrollable?: boolean;
+    defaultText?: string;
+    options: T[];
+    displayPath?: string;
+    serializationPath?: string | null;
+  };
+  Blocks: {
+    display?: [T | undefined];
+    option?: [T | undefined];
+    empty?: [];
+  };
+  Element: HTMLButtonElement;
+}
+
+const baseDefaultText = 'Select an Option'; // TODO i18n
```

Comment:
Since i18n is now included, could you go ahead and handle that? I'll get a PR together that'll disallow "bare strings" and require everything to go through `ember-intl` in the future.